h2o-algos/src/main/java/hex/gam/GAM.java

```diff
@@ -70,15 +82,46 @@ public GAM(GAMModel.GAMParameters parms, Key<GAMModel> key) {
     super(parms, key);
     init(false);
   }
-  
-  double[][] _knots;
 
-  /***
-   * This method will look at the keys of knots stored in _parms._knot_ids and copy them over to double[][]
-   * array.
-   *
-   * @return double[][] array containing the knots specified by users
-   */
+  // cross validation can be used to choose the best alpha/lambda values among a whole collection of alpha
+  // and lambda values.  Future hyperparameters can be added for cross-validation to choose as well.
+  @Override
+  public void computeCrossValidation() {
+    _cv_on = true;  // set cross-validation on to be true
+    _validKeys = new IcedHashSet<>();
+    if (error_count() > 0)
+      throw H2OModelBuilderIllegalArgumentException.makeFromBuilder(GAM.this);
+    super.computeCrossValidation();
+  }
+
+  // find the best alpha/lambda values used to build the main model moving forward by looking at the devianceValid
+  @Override
+  public void cv_computeAndSetOptimalParameters(ModelBuilder[] cvModelBuilders) {
```

Comment:
Same problem with CamelCase here. It is hard to read underscore in a Java method. :( But if there is a rule use cross-validation methods like this, keep it as is.

Reply:
Yeah, I am overriding an existing method.  I agree with you about the name issues though.